splinepy/helpme/extract.py

```diff
@@ -449,65 +449,70 @@ def spline(spline, para_dim, split_plane):
         split_plane = [split_plane]
 
     # Check if is bezier-type
-    is_bezier = "Bezier" in spline.whatami
-    is_rational = "weights" in spline.required_properties
-    if is_bezier:
-        spline_copy = spline.nurbs if is_rational else spline.bspline
+    if not spline.has_knot_vectors:
+        spline_copy = spline.nurbs if spline.is_rational else spline.bspline
```

Comment:
This is totally fine. I just wanna let you know / note that we can maybe extend something like [this](https://tataratat.github.io/splinepy/_generated/splinepy.splinepy_core.same_spline_with_knot_vectors.html#splinepy.splinepy_core.same_spline_with_knot_vectors) in `create`

Reply:
Next PR